keras_cv/layers/preprocessing/rgb_shift.py

```diff
@@ -0,0 +1,144 @@
+        self.seed = seed
+
+    def _set_shift_limit(self, shift_limit, channel=''):
+        if isinstance(shift_limit, (tuple, list)):
+            if len(shift_limit) != 2: 
+                raise ValueError(f'The {channel} should be scalar, tuple or list of two upper and lower \
+                    bound number. Got {shift_limit}')
+            return sorted(shift_limit)
+        elif isinstance(shift_limit, (int, float)):
+            shift_limit = abs(shift_limit)
+            return [-shift_limit, shift_limit]
+        else:
+            raise ValueError(f'The {channel} should be scalar, tuple or list of two upper and lower bound\
+                 number. Got {shift_limit}')
+
+    @tf.function
+    def _rgb_shifting(self, image):
+        r_shift = tf.random.uniform([], 
+                                    minval=self._r_shift_limit[0], 
+                                    maxval=self._r_shift_limit[1], 
+                                    dtype=tf.float32, seed=self.seed)
+        g_shift = tf.random.uniform([], 
+                                    minval=self._g_shift_limit[0], 
+                                    maxval=self._g_shift_limit[1], 
+                                    dtype=tf.float32, seed=self.seed)
+        b_shift = tf.random.uniform([], 
+                                    minval=self._b_shift_limit[0],
+                                    maxval=self._b_shift_limit[1], 
+                                    dtype=tf.float32, seed=self.seed)
+        
+        unstack_rgb = tf.unstack(image, axis=-1)
+        shifted_rgb = tf.stack([unstack_rgb[0] + r_shift, 
+                                unstack_rgb[1] + g_shift, 
+                                unstack_rgb[2] + b_shift],  axis=-1)
+        shifted_rgb = tf.clip_by_value(shifted_rgb, 0, 255)
+        return tf.cast(shifted_rgb, image.dtype)
+
+    def call(self, images, training=None):
+        """call method for the RGBShift layer.
+
+        Args:
+            images: Tensor representing images of shape
+                [batch_size, width, height, channels], with dtype tf.float32 / tf.uint8, or,
+                [width, height, channels], with dtype tf.float32 / tf.uint8
+        Returns:
+            images: augmented images, same shape as input.
+        """
+        if training is None:
+            training = backend.learning_phase()
+
+        if not training:
+            return images
+        else:
+            unbatched = images.shape.rank == 3
+            # The transform op only accepts rank 4 inputs, so if we have an unbatched
+            # image, we need to temporarily expand dims to a batch.
+            if unbatched:
+                images = tf.expand_dims(images, axis=0)
+
+            # TODO: Check if input tensor is RGB or not. 
+            # TODO: Support tf.uint8
+
+            # TODO: Make the batch operation vectorize.
+            output = tf.map_fn(lambda image: self._rgb_shifting(image), images)
```

Comment:
I think we should fix the batch verctorization within this PR. Eg you can generate the shift based on teh input rank. Use tf.map will have a performance implication.

Reply:
Done.